PMMain/Midas/MidasGUI/RingData.cpp

```diff
@@ -0,0 +1,31 @@
+#include "RingData.h"
+
+
+ringData::ringData()
+{
+}
+
+
+ringData::~ringData()
+{
+}
+
+std::vector<ringData::keyboard_value> ringData::getRingInVector()
+{
+    return*ringInVector;
+}
+
+std::vector<ringData::keyboard_value> ringData::getRingOutVector()
+{
+    return*ringOutVector;
+}
+
+void ringData::setRingInVector(std::vector<keyboard_value> &ringInVal)
+{
+    ringInVector = &ringInVal;
```

Comment:
This seems a bit strange. Why are you storing a handle to what the caller passes in instead of just copying the data?